src/query/functions-v2/src/scalars/string_multi_args.rs

```diff
@@ -348,3 +679,515 @@ fn char_fn(args: &[ValueRef<AnyType>], _: &GenericMap) -> Result<Value<AnyType>,
     };
     Ok(Value::Column(Column::String(result)))
 }
+
+fn regexp_instr_fn(args: &[ValueRef<AnyType>], _: &GenericMap) -> Result<Value<AnyType>, String> {
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+
+    let source_arg = args[0].try_downcast::<StringType>().unwrap();
+    let pat_arg = args[1].try_downcast::<StringType>().unwrap();
+    let pos_arg = if args.len() >= 3 {
+        Some(args[2].try_downcast::<Int64Type>().unwrap())
+    } else {
+        None
+    };
+    let occur_arg = if args.len() >= 4 {
+        Some(args[3].try_downcast::<Int64Type>().unwrap())
+    } else {
+        None
+    };
+    let ro_arg = if args.len() >= 5 {
+        Some(args[4].try_downcast::<Int64Type>().unwrap())
+    } else {
+        None
+    };
+    let mt_arg = if args.len() >= 6 {
+        Some(args[5].try_downcast::<StringType>().unwrap())
+    } else {
+        None
+    };
+
+    let size = len.unwrap_or(1);
+    let mut key: Vec<Vec<u8>> = Vec::new();
+    let mut map: HashMap<Vec<Vec<u8>>, Regex> = HashMap::new();
+    let mut builder = NumberColumnBuilder::with_capacity(&NumberDataType::UInt64, size);
+    for idx in 0..size {
+        unsafe {
+            let source = source_arg.index_unchecked(idx);
+            let pat = pat_arg.index_unchecked(idx);
```

Comment:
Are we going to cache the regex for scalar pat later?

Reply:
`regex` is cached in a map, if `pat` and `match type` is equal.